src/bidiMapper/domains/script/scriptEvaluator.ts

```diff
@@ -317,29 +63,317 @@ export async function stringifyObject(
+            awaitPromise: false,
+            arguments: args,
+            returnByValue: false,
+            executionContextId: realm.executionContextId,
+          }
+        );
+        // TODO: release `argEvalResult.result.objectId`  after using.
+        // https://github.com/GoogleChromeLabs/chromium-bidi/issues/375
+        return {objectId: argEvalResult.result.objectId};
+      }
+      case 'set': {
+        // TODO(sadym): if non of the nested items has remote reference,
+        //  serialize to `unserializableValue` without CDP roundtrip.
+        const args = await this.#flattenValueList(argumentValue.value, realm);
+
+        const argEvalResult = await realm.cdpClient.sendCommand(
+          'Runtime.callFunctionOn',
+          {
+            functionDeclaration: String((...args: unknown[]) => {
+              return new Set(args);
+            }),
+            awaitPromise: false,
+            arguments: args,
+            returnByValue: false,
+            executionContextId: realm.executionContextId,
+          }
+        );
+        // TODO: release `argEvalResult.result.objectId`  after using.
+        // https://github.com/GoogleChromeLabs/chromium-bidi/issues/375
+        return {objectId: argEvalResult.result.objectId};
+      }
+
+      case 'PROTO.binding': {
+        const bindingName = argumentValue.value;
+
+        const argEvalResult = await realm.cdpClient.sendCommand(
+          'Runtime.callFunctionOn',
+          {
+            functionDeclaration: String(() => {
+              const queue: string[] = [];
+              let resolver: (_: string | PromiseLike<string>) => void;
+
+              return {
+                /**
+                 * Gets a promise, which is resolved as soon as a message occurs
+                 * in the queue.
+                 */
+                async getMessage(): Promise<string> {
+                  if (queue.length > 0) {
+                    return queue.shift()!;
+                  }
+                  const promise = new Promise<string>((resolve) => {
+                    resolver = resolve;
+                  });
+                  promise.then(() => queue.shift());
+                  return promise;
+                },
+
+                /**
+                 * Adds a message to the queue.
+                 * If the queue is initially empty, resolve the pending promise.
+                 */
+                sendMessage(message: string) {
+                  queue.push(message);
+                  if (message.length === 1) {
+                    resolver(message);
+                  }
+                },
+              };
+            }),
+            returnByValue: false,
+            executionContextId: realm.executionContextId,
+            generateWebDriverValue: false,
+          }
+        );
+        const bindingHandle = argEvalResult.result.objectId;
+
+        // Long-poll the message queue.
+        // TODO(BLOCKER): When do we break from the loop?
```

Comment:
We will never return `sendMessage` if we don't break here.

Reply:
you can run async code without awaiting for the result:
```suggestion
     (async ()=>{
       while(true) {
         await cdp.callFunctionOn({
             functionDeclaration: "this.getMessage()", 
             objectId: bindingHandle,
             awaitPromise: true
         });
       }
     })()
```

Reply:
Done